app/controllers/schools/base_controller.rb

```diff
@@ -1,45 +1,16 @@
 module Schools
   class BaseController < ApplicationController
-    def current_user
-      @current_user ||= User.from_json SAMPLE_USER_JSON
-    end
+    include DFEAuthentication
+    before_action :require_auth
 
-    def current_school
-      @current_school ||= School.from_json SAMPLE_SCHOOL_JSON
-    end
+    rescue_from ActionController::ParameterMissing, with: -> { redirect_to schools_errors_no_school_path }
+    rescue_from ActiveRecord::RecordNotFound, with: -> { redirect_to schools_errors_not_registered_path }
```

Comment:
I'm pretty sure these are rescuing in the wrong place - this will affect anything deriving from `Schools::BaseController`

In effect anything else which triggers a ParameterMissing or RecordNotFound will end up in these redirects - instead of hitting the normal handlers.

The actual points you want rescuing should be handle the rescue instead9